lib/rabbit_feed/connection.rb

```diff
@@ -0,0 +1,22 @@
+module RabbitFeed
+  class Connection
+    include Singleton
+
+    def initialize
+      @connection = Bunny.new RabbitFeed.configuration.connection_options
+      @connection.start
+      @channel = @connection.create_channel
+      @mutex = Mutex.new
+    end
+
+    protected
+
+    attr_reader :channel
+
+    def thread_safe &block
```

Comment:
yeah `synchronized` is fine... is the Java name :smile: ... why is this method protected and not private?